### PR TITLE
Ports the Contender from old codebase.

### PR DESCRIPTION
**Pull request by Manly-Dwarf**
[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. This includes, new features, sprites, sounds, balance changes, admin tools, map edits, removals, big refactors, config changes, hosting changes and important fixes. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs)

:cl: KawaiiBigBoss
add: Adds a 7 TC traitor item called the Contender, a double barrel shotgun which can hold any ammo you put into it.
add: Also added a 13 TC "Bucket of Buckshot", because the Contender could use it as well as the PR I'm making right after this one.
/:cl:

The Contender is a 7 TC traitor item that was in our original codebase. Made by ArcLumin, it could hold any ammo you could insert into it - even Strider MGL grenades.
Also adds the Box of Buckshot, because it's either this PR or the next one. It is necessary, I assure you.

### CODE REVIEW

File: code/modules/projectiles/box_magazine.dm [hunk 1]
```diff
@@ -40,10 +40,17 @@
...
+	if ((!R || (caliber && R.caliber != caliber) || (!caliber && R.type != ammo_type)) && (caliber != "all")) /* hippie edit end */
 		return 0
 
+/*hippie edit */
+	if ((caliber == "all") && (stored_ammo.len < max_ammo))
+		stored_ammo += R
+		R.loc = src
+		return 1
+/*hippie edit end*/
+
```
Comment (Manly-Dwarf) [inline code/modules/projectiles/box_magazine.dm]: Yes, this is necessary for it to work. I can't mirror the proc.

**Comment by ghost** [comment]
yea i loved this weapon +1

File: code/modules/projectiles/box_magazine.dm (hunk 1 above)
Comment (Carbonhell) [reply code/modules/projectiles/box_magazine.dm]: Are you sure? What happens if you do not put this if block?

**Comment by ghost** [comment]
what does the bucket of buckshot do?

**Comment by Manly-Dwarf** [comment]
Bucket of Buckshot is literally just a box that contains 60 shots (13 TC's worth - if we're going by how much a standard drum magazine costs) of Buckshot.
It was requested a long time ago. Considering what item I'll be able to port in next once its in, I don't think it's too far fetched to include them both in a single PR.

File: code/modules/projectiles/box_magazine.dm (hunk 1 above)
Comment (Manly-Dwarf) [reply code/modules/projectiles/box_magazine.dm]: What happens is it don't do nuffin.

**Comment by Carbonhell** [comment]
+1, only needs a codecheck now

**Comment by 2Dollarbill** [comment]
+1, more toys for traitors is always gucci.

File: code/modules/projectiles/box_magazine.dm [hunk 2]
```diff
@@ -40,10 +40,17 @@
...
-	// Boxes don't have a caliber type, magazines do. Not sure if it's intended or not, but if we fail to find a caliber, then we fall back to ammo_type.
-	if(!R || (caliber && R.caliber != caliber) || (!caliber && R.type != ammo_type))
+	// Boxes don't have a caliber type, magazines do. Not sure if it's intended or not, but if we fail to find a caliber, then we fall back to ammo_type. - next line contains a hippie edit, && caliber!= "all"
+	if ((!R || (caliber && R.caliber != caliber) || (!caliber && R.type != ammo_type)) && (caliber != "all")) /* hippie edit end */
 		return 0
 
+/*hippie edit */
+	if ((caliber == "all") && (stored_ammo.len < max_ammo))
+		stored_ammo += R
+		R.loc = src
```
Comment (tortellinitony) [inline code/modules/projectiles/box_magazine.dm]: use forcemove

**Review by tortellinitony — DISMISSED** [review]

**Comment by JohnMcLaime** [comment]
60 isnt worth that much tc, make it infinite buckshot, since it is worse than beanbag shells.

**Comment by Manly-Dwarf** [comment]
Actually 60 is worth exactly 13 TC based on how much the drum magazine costs. 2 TC for 8 shots.
On top of that, beanbags and rubber shots are used more by crew members and not Nuke Ops - as Nuke Ops use stun slugs instead.

**Comment by Manly-Dwarf** [comment]
Used forcemove. If this breaks everything I blame Tony.

**Comment by JohnMcLaime** [comment]
Drug mag is an extra mag, even if you run out of it, you should still keep it, since loading the mag to full takes 1 click on the ammo pile.
Most shotguns are meh weapons, even techshells are underwhelming. Shotguns are pain in the ass to maintain with their only 1 ammo per click loading, just make buckshot bucket refill every x time, like holopucks do. 
Why would you spend so much tc on this when you can just steal autolathe and make shitton of basically free shells or just buy revolver and steal autolathe for same tc.

**Comment by Manly-Dwarf** [comment]
@JohnMcLaime Because of nuke ops. The box of buckshot is here for the contender, and for the future. I explained this in the PR.
If it turns out to be underpowered, we can tune it in the future.
The Contender never was a very powerful weapon to begin with. Making the box considerably cheaper is not a way to make the contender better.
What price do you honestly recommend for the box?

**Comment by JohnMcLaime** [comment]
Box or bucket? For 60 shells, i doubt i would buy it even for 5 tc (just restrict from nukeops).  Why not make it replenishable, maybe hit drunk people with it to create shells or something silly like that.
Finite items like this, which can easily be replaced for no/less tc (emag) is bad balancing in my book, bartenders weapons are bad for this type of item anyway (buy and go ham right away)

**Comment by Manly-Dwarf** [comment]
Why would I restrict an item designed for Nuke Ops from Nuke Ops?
Making it refillable but have less starting ammo is something I might do in the future. How about 7 TC instead of 13?

**Comment by JohnMcLaime** [comment]
Ah i thought it was for barman (since they had a retarded overpriced buckshot ammo box prebase at some point).
Dont nukies already have surplus ammo bag for shotguns? If you wanna make it buckshot only with no extra mags, better up the ammount of ammo by 40% (if not more) compared to your original pricing.
Also what is the size of the item?

**Review by GuyonBroadway — APPROVED** [review]
dis luks gud.

**Comment by GuyonBroadway** [comment]
Has the +1's and time necessary.

**Comment by ghost** [comment]
Your Fix/Feature pull request delta is currently below zero (-1). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Comment by GuyonBroadway** [comment]
Nobody gives a fuck.

**Pull request merged**